packages/storage/src/implementation/requests.ts

```diff
@@ -279,6 +279,41 @@ export function metadataForUpload_(
   return metadataClone;
 }
 
+/**
+ * Prepare RequestInfo for non-resumable uploads as Content-Type: octet-stream.
+ */
+export function simpleUpload(
+  service: StorageService,
+  location: Location,
+  mappings: MetadataUtils.Mappings,
+  blob: FbsBlob,
+  metadata?: Metadata | null
+): RequestInfo<Metadata> {
+  const urlPart = location.bucketOnlyServerUrl();
+  const headers: { [prop: string]: string } = {
+    'Content-Type': 'application/octet-stream'
```

Comment:
Normally (in the old resumable setup) the content type in the header doesn't actually match the metadata.contentType field because the metadata has a specific content type as seen in the logic above, and the request header Content-Type is always `multipart`. Since we're doing away with `multipart` I guess it would be best to make the header match the metadata.